### PR TITLE
Adding Napa County

### PR DESCRIPTION
**Pull request by miloconway**
Following the same format as that done for Solano County. However ran into a wrinkle - the coordinate system from the shp file doesn't seem to be long-lat and not quite sure how to work with it.

The input data file contains an associated XML file for the SHP that has the following:

```
<coordRef>
<type Sync="TRUE">Projected</type>
<geogcsn Sync="TRUE">GCS_North_American_1983</geogcsn>
<csUnits Sync="TRUE">Linear Unit: Foot_US (0.304801)</csUnits>
<projcsn Sync="TRUE">NAD_1983_StatePlane_California_II_FIPS_0402_Feet</projcsn>
<peXml Sync="TRUE"><ProjectedCoordinateSystem xsi:type='typens:ProjectedCoordinateSystem' xmlns:xsi='http://www.w3.org/2001/XMLSchema-instance' xmlns:xs='http://www.w3.org/2001/XMLSchema' xmlns:typens='http://www.esri.com/schemas/ArcGIS/10.7'><WKT>PROJCS[&quot;NAD_1983_StatePlane_California_II_FIPS_0402_Feet&quot;,GEOGCS[&quot;GCS_North_American_1983&quot;,DATUM[&quot;D_North_American_1983&quot;,SPHEROID[&quot;GRS_1980&quot;,6378137.0,298.257222101]],PRIMEM[&quot;Greenwich&quot;,0.0],UNIT[&quot;Degree&quot;,0.0174532925199433]],PROJECTION[&quot;Lambert_Conformal_Conic&quot;],PARAMETER[&quot;False_Easting&quot;,6561666.666666666],PARAMETER[&quot;False_Northing&quot;,1640416.666666667],PARAMETER[&quot;Central_Meridian&quot;,-122.0],PARAMETER[&quot;Standard_Parallel_1&quot;,38.33333333333334],PARAMETER[&quot;Standard_Parallel_2&quot;,39.83333333333334],PARAMETER[&quot;Latitude_Of_Origin&quot;,37.66666666666666],UNIT[&quot;Foot_US&quot;,0.3048006096012192],AUTHORITY[&quot;EPSG&quot;,2226]]</WKT><XOrigin>-115211800</XOrigin><YOrigin>-93821500</YOrigin><XYScale>36983428.057351544</XYScale><ZOrigin>-100000</ZOrigin><ZScale>10000</ZScale><MOrigin>-100000</MOrigin><MScale>10000</MScale><XYTolerance>0.0032808333333333331</XYTolerance><ZTolerance>0.001</ZTolerance><MTolerance>0.001</MTolerance><HighPrecision>true</HighPrecision><WKID>102642</WKID><LatestWKID>2226</LatestWKID></ProjectedCoordinateSystem></peXml>
</coordRef>
```

This probably contains the info needed to translate the coordinates but not sure how to use it

### CODE REVIEW

**Comment by miloconway** [comment]
OK, figured out the coordinate system translation, updated.

Generated file: https://drive.google.com/file/d/14ec-92Fmkn-VL5QrzL3T0tDVMZ2_2tjN/view?usp=sharing

File: app/index.html [hunk 1]
```diff
@@ -431,6 +431,10 @@
             countyStr = 'Marin';
             disclaimer = `<br><br><strong>Warning</strong>: Due to a scraping issue, Marin properties sold within the last year are displaying incorrectly with low tax.  This issue affects a very small number of properties.  Please check the tax records before sharing.  APN ${record.apn}`;
             break;
+          case 'NAPA':
+            taxUrl = `https://common2.mptsweb.com/MBC/napa/tax/main/${record.apn}/2020/0000`;
```
Comment (rgbkrk) [inline app/index.html:435]: Is the `2020` here the year? Do we need to have this get replaced with the current year?

File: scrapers/napa/parse.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+import sys
+
+from shapely.geometry import Polygon
+
+csv.field_size_limit(sys.maxsize)
+
+ADDRESS_REGEX = re.compile('Address.*\n[^.*\>]+\>([^\<]+).*\n.*\n[^.*\>]+\>([^\<]+)')
+AMOUNTS_REGEX = re.compile('h4\>Totals\s\-.*\n.*\n.*\n.*\>Total Due\<.*\n[^\>]+\>([^\<]+)')
+GEOJSON_FILE = '/home/alin/Downloads/napa/napa.geojson'
+OUTPUT_FILE = '/home/alin/code/prop13/scrapers/napa/parse_output.csv'
```
Comment (rgbkrk) [inline scrapers/napa/parse.py]: We probably don't want to merge this in with your paths, so here's a proposal:

* Create an `outputs` directory at the root of the repo with a README that gitignores everything else put in it
* The README states that this directory is used only for localized runs and that data should not be included
* We calculate the path for the outputs like this in the script:

```python
import os 
output_path = os.path.join("..", "..", "outputs", "napa")
os.mkdir(output_path)
OUTPUT_FILE = os.path.join(output_path, "parse_output.csv")
```

File: scrapers/napa/scrape.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import csv
+import os
+import time
+
+import requests
+
+CONNECTIONS = 20
+PARCEL_LOOKUP_URL = 'https://common2.mptsweb.com/MBC/napa/tax/main/%s/2020/0000'
+PARCEL_SOURCE_FILE = '/home/alin/Downloads/napa/parcels_public.csv'
+OUTPUT_DIR = '/home/alin/code/prop13/scrapers/napa/scrape_output'
```
Comment (rgbkrk) [inline scrapers/napa/scrape.py]: Same statement as above about putting the files in a canonical location for repeatability.

File: app/index.html (hunk 1 above)
Comment (miloconway) [reply app/index.html:435]: 2020 is the year, and it is the latest record the county has

File: scrapers/napa/parse.py (hunk 2 above)
Comment (miloconway) [reply scrapers/napa/parse.py]: Good suggestion! I've done it for this county, and also applied it to solano county which I had previously worked on

**Comment by typpo** [comment]
This is looking great!

The XY coordinates are [CA State Plane coordinates](https://www.conservation.ca.gov/cgs/Pages/Program-RGMP/california-state-plane-coordinate-system.aspx) at Napa is in Zone 2 ([ESPG 2226](https://epsg.io/2226)):

```python
from pyproj import Transformer

# California Zone 2
# https://epsg.io/2226
transformer = Transformer.from_crs(2226, 4326)

# ....later on in your parcel loop....
lat, lng = transformer.transform(y_coord, x_coord)
```

You may have to swap x/y lng/lat.  I've seen the order of the coordinate vary between datasources.

Hope this helps :+1:

**Comment by miloconway** [comment]
@typpo yup! i examined the generated geojson output for the default output and previously working samples and figured out how to get it working, the code and the linked g-drive data in the subsequent comment should work!

**Comment by typpo** [comment]
Oops, I totally missed that.  Merging & updating the data now!

**Pull request merged**